### PR TITLE
HMD and Away improvements

### PR DESCRIPTION
**Pull request by ZappoMan**
- ESC now enters away mode.
- When in away mode, the mouse is released so you can see and use the system mouse.
- when in away mode, the mouse is hidden in the HMD so that you know it's not active
- when returning from away mode, we automatically center the pointer if in HMD
- the "Paused" message when in away mode, will follow your view in the HMD

### CODE REVIEW

**Comment by ZappoMan** [comment]
Fixes this bug - https://app.asana.com/0/26320649018758/94455201563172

**Comment by hifi-gustavo** [comment]
Build console output for commit: 95c92f7a2793b47dc928120430cd6f8bec88ed24 available at: https://gist.github.com/4142bf99457a7d0f2a4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 501d23021ffef607d83e75c7c15916dfd2c90f21 available at: https://gist.github.com/4142bf99457a7d0f2a4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.dmg

**Comment by ZappoMan** [comment]
Fixes these bugs
- https://app.asana.com/0/26320649018758/94455201563172
- https://app.asana.com/0/26320649018758/95036254781069

**Comment by hifi-gustavo** [comment]
Build console output for commit: 52c38e56a493b0a1b02457808e19ca35ad7e4e8c available at: https://gist.github.com/4142bf99457a7d0f2a4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.dmg

File: examples/away.js [hunk 1]
```diff
@@ -64,29 +79,42 @@ function stopAwayAnimation() {
...
 function showOverlay() {
-    var properties = {visible: true},
+    var properties = {visible: true};
+
+    if (HMD.active) {
+        Overlays.editOverlay(overlayHMD, { visible: true, position : HMD.getHUDLookAtPosition3D() });
+    } else {
         // Update for current screen size, keeping overlay proportions constant.
-        screen = Controller.getViewportDimensions(),
+        var screen = Controller.getViewportDimensions(),
```
Comment (hyperlogic) [inline examples/away.js]: Good catch

Comment (ZappoMan) [inline examples/away.js]: actually, that change wasn't a catch per se... it was just that I was moving the other variable before it to a different scope, so I needed to add back the `var` that moved.

File: examples/away.js [hunk 2]
```diff
@@ -64,29 +81,74 @@ function stopAwayAnimation() {
...
-        properties.y = 0;
-        properties.x = (screen.x - properties.width) / 2;
+        // make sure HMD is hidden
+        Overlays.editOverlay(overlayHMD, { visible: false });
+
+        // Update for current screen size, keeping overlay proportions constant.
+        var screen = Controller.getViewportDimensions();
+
+        // keep the overlay it's natural size and always center it...
+        Overlays.editOverlay(overlay, { visible: true,
```
Comment (ZappoMan) [inline examples/away.js]: NOTE: there's apparently a bug in 2D overlays that you can't change their size after they've been created (and/or something else subtle going on)... and so the old logic which attempted to stretch or shrink the PAUSED graphic wasn't working correctly. This is a temporary solution to always center the PAUSED message in the center of the 2D display and keep it it's natural size... we should revisit this after fixing the other overlay bugs.

**Comment by ZappoMan** [comment]
@cozza13 @AndrewMeadows - it's ready for review again...

Known issues:
- Something was broken in the old away logic's attempt to center/shrink/stretch the PAUSED message, I wasn't able to get it to work like the original author intended, so I changed the behavior slightly.
- It turns out that in 3D/HMD mode, although we're rendering the PAUSED message between the viewer and the HUD, the HUD is actually rendered "after" the PAUSED message, this means that basically, if there's a 2D window over the PAUSED message, the PAUSED message will be somewhat obscured by the 2D window. There's not much I can do about that without major refactoring, so I don't think it's a pre-MVP issue we want to address.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ba7b5c6e23529c540242a4c8377c3200c7883c2d available at: https://gist.github.com/4142bf99457a7d0f2a4c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7228.dmg

**Comment by cozza13** [comment]
tested and it works

File: examples/away.js (hunk 2 above)
Comment (AndrewMeadows) [inline examples/away.js]: Is there an asana task about this 2D overlay bug?  I coudn't find it.

Comment (jherico) [inline examples/away.js]: maybe just make it a QML window?

**Pull request merged**